crates/typst/src/visualize/image/mod.rs

```diff
@@ -190,6 +191,21 @@ fn layout_image(
         Smart::Auto => determine_format(elem.path().as_str(), data).at(span)?,
     };
 
+    // Warn the user if the image contains a foreign object. Not perfect
+    // because the svg could also be encoded, but that's an edge case.
+    if format == ImageFormat::Vector(VectorFormat::Svg) {
+        let needle = b"<foreignObject";
+        if data
+            .as_slice()
+            .windows(needle.len())
+            .any(|s| s == needle)
```

Comment:
This seems potentially rather slow. I think `std::str::from_utf8` + `contains` might actually be faster because `str::contains` uses an optimized string search algorithm. We could also use https://docs.rs/memmem but it feels like overkill to pull a crate.

Reply:
We could have `Readable::as_str -> Option<&str>` to do this. Skips the UTF-8 check if its already a string.

Reply:
Yes, I had the same thought, will change that.